src/smartpeak/source/ui/FilePicker.cpp

```diff
@@ -123,27 +242,45 @@ namespace SmartPeak
         ImGui::TableNextRow();
         for (int column = 0; column < column_count; column++)
         {
-          char buf[256];
-          sprintf(buf, "%s", pathname_content_[column][row].c_str());
+          ImDirectoryEntry* item = &Im_directory_entries[row];
+          char text_buffer[256];
+          if (column == 0)
+          {
+            sprintf(text_buffer, "%s", item->Name);
+          }
+          else if (column == 1)
+          {
+            sprintf(text_buffer, "%s", item->Size);
+          }
+          else if (column == 2)
+          {
+            sprintf(text_buffer, "%s", item->Type);
+          }
+          else if (column == 3)
+          {
+            sprintf(text_buffer, "%s", item->DateModified);
+          }
+          
           const bool is_selected = (selected_entry == row);
           ImGui::TableSetColumnIndex(column);
-          if(ImGui::Selectable(buf, is_selected, selectable_flags))
+          if(ImGui::Selectable(text_buffer, is_selected, selectable_flags))
           {
             selected_entry = row;
-            std::strcpy(selected_filename, pathname_content_[0][selected_entry].c_str());
-            if (ImGui::IsMouseDoubleClicked(0) && pathname_content_[2][selected_entry] == "Directory") // TODO: error prone to rely on strings
+            std::strcpy(selected_filename, Im_directory_entries[selected_entry].Name);
```

Comment:
Is this the potential cause of the duplication when selecting the filename in the FilePicker?

Reply:
It was because the `picked_pathname_.append()` was called twice in `ImGui::Button("Open")` block